cpp/dolfinx/refinement/plaza.cpp

```diff
@@ -333,10 +328,16 @@ face_long_edge(const mesh::Mesh& mesh)
     assert(it1 != cell_vertices.end());
     const std::size_t local1 = std::distance(cell_vertices.begin(), it1);
 
-    auto x_dofs = x_dofmap.links(c);
-    auto x0 = xt::row(x, x_dofs[local0]);
-    auto x1 = xt::row(x, x_dofs[local1]);
-    edge_length[e] = xt::norm_l2(x0 - x1)();
+    auto x_dofs = x_dofmap.links(cells.front());
+    xtl::span<const double, 3> x0(
+        mesh.geometry().x().data() + 3 * x_dofs[local0], 3);
+    xtl::span<const double, 3> x1(
+        mesh.geometry().x().data() + 3 * x_dofs[local1], 3);
+
+    // Compute length of edge between vertex x0 and x1
+    edge_length[e] = std::sqrt(std::transform_reduce(
+        x0.begin(), x0.end(), x1.begin(), 0.0, std::plus<>(),
+        [](auto x0, auto x1) { return (x0 - x1) * (x0 - x1); }));
```

Comment:
Can we skip the `sqrt` ?

Reply:
> Can we skip the `sqrt` ?

Is it not needed?

Reply:
I dont think so, it is just a measure of the longest edge. I guess this would just change the quality ratio down the line when deciding if we do uniform refinement of a cell in 2D

Reply:
Yes, I suppose so. But maybe we should just remove this feature, as it is not very important, adds complexity, and only works in 2D.